### ISSUE TITLE
Manual install throws various errors in Amazon Linux 2

### ISSUE THREAD

**Issue by gamefundas** [netdata/netdata#9208]
Netdata 1.22.1 
OS - Amazon Linux 2 (CentOS/RHEL)

I am dealing with too many installation issues with newer versions of Netdata. Upgrading has become a huge hurdle recently.

After trying the gz.run file which works fine but couldnt get ebpf collector running I tried the manual install and here I starting seeing various other problems.

1. dashboard.tar fails with following errors, I dont know what this means.
dashboard.tar.gz: FAILED
sha256sum: WARNING: 1 computed checksum did NOT match
 FAILED  dashboard files checksum validation failed. 

 FAILED  Unable to fetch React dashboard. The install process will continue, but you will not be able to use the new dashboard.

2. Go plugin installation fails - Dont remember seeing this problem when I was in Netdata 1.20. Why download this from outside, since this is a bundle why not include it as we dont have access to any form of internet in our VPC. If this cant be pacakges then there should be an option to source it from local file system. I would hate to lose all go plugins.

go.d.plugin-v0.18.0.linux-amd64.tar.gz: FAILED
config.tar.gz: FAILED
sha256sum: WARNING: 2 computed checksums did NOT match
go.d plugin checksum validation failure.
Either check the error or consider disabling it by issuing '--disable-go' in the installer

 FAILED  go.d.plugin package files checksum validation failed.

3. ebpf collector failure - I am upgrading mainly for this. This seems to be sourcing the script again from the internet.

--- eBPF Kernel Collector (opt-in) --- 
 Using eBPF Kernel Package built against Linux 4.14.171
 Downloading check-kernel-config.sh ...
curl: (56) Received HTTP code 403 from proxy after CONNECT
 FAILED  Failed to download check-kernel-config.sh 

./netdata-installer.sh: line 1370:  Removing temporary directory /tmp/netdata-ebpf-CzxmwH ...: ambiguous redirect

Is there any other way for me to install the latest versions without internet access. RPM's have failed me as well since I dont have the luxury of modifying our yum repos.

If there is a way for me to enable ebpf using the gz.run bundle it might be the easiest at this point. I have another issue opened for that failure.

**Comment by underhood** [comment]
Hello there, sad to hear netdata is giving you trouble,

as you already seem to have found out the installer needs access to internet to download some archives to work.
Since you mention the system doesn't have internet access did you try to take a look at the docu for Offline installation documentation: https://learn.netdata.cloud/docs/agent/packaging/installer/methods/offline

**Comment by gamefundas** [comment]
Offline install using local files as much as I know are the tar.gz and gz.run methods behind the scenes wrapped with the kickstart-static64.sh script. Using the tar method leads to lot of undesirables as we noticed in the issue described above. With the gz.run I am able to install but ebpf mode cant be activated due to some missing libraries as part of the bundle which is I opened another issue #9201

**Comment by thiagoftsm** [comment]
Hi @gamefundas ,

I was reading your message and I detected the possible problem that are not allowing to install:

1 - You have an installation that to d download of  kernel 4.14, please, can you run the following commands and give us the output? I want to confirm that you are using the kernel `3.10.0-x`. 

```bash
uname -r
uname -v
cat /etc/redhat-release
```

About the issue #9201, I gave an answer for you there, if you are using CentOS 6.5 unfortunately you cannot run eBPF collector, because the kernel version is [2.6.32-431](http://vault.centos.org/6.5/updates/Source/SPackages/) according the CentOS link. As I explained in the issue #9201, the eBPF is newer than any kernel 2.6.x, in reality the features that Netdata uses were created on kernel 3.18.0. 

We can run the ebpf collector on the latest CentOS 7.x, because RedHat created patches for the kernels 3.10. 

Best regards!

**Comment by prologic** [comment]
> 1. Dont remember seeing this problem when I was in Netdata 1.20. Why download this from outside, since this is a bundle why not include it as we dont have access to any form of internet in our VPC. If this cant be pacakges then there should be an option to source it from local file system. I would hate to lose all go plugins.

@gamefundas This has always been the case since I joined the company at least. But you are right; we can and probably should just bundle everything into a single source dist tarball. We do for the static prebuilt Netdata for x86_64 anyway (_obviously_), no reason we can't bundle all required sources for the source dist.

However is there any particular reason you want to compile from source and not just use the static install? It should work even without eBPF support... Please let me know if this is causing you any trouble?

**Comment by gamefundas** [comment]
cat /proc/version
Linux version 4.14.154-128.181.amzn2.x86_64 (mockbuild@ip-10-0-1-129) (gcc version 7.3.1 20180712 (Red Hat 7.3.1-6) (GCC)) #1 SMP Sat Nov 16 21:49:00 UTC 2019

**Comment by gamefundas** [comment]
And some more info...

cat /etc/*elease
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"
Amazon Linux release 2 (Karoo)

**Comment by gamefundas** [comment]
> > 1. Dont remember seeing this problem when I was in Netdata 1.20. Why download this from outside, since this is a bundle why not include it as we dont have access to any form of internet in our VPC. If this cant be pacakges then there should be an option to source it from local file system. I would hate to lose all go plugins.
> 
> @gamefundas This has always been the case since I joined the company at least. But you are right; we can and probably should just bundle everything into a single source dist tarball. We do for the static prebuilt Netdata for x86_64 anyway (_obviously_), no reason we can't bundle all required sources for the source dist.
> 
> However is there any particular reason you want to compile from source and not just use the static install? It should work even without eBPF support... Please let me know if this is causing you any trouble?

Yea guess I will have to settle down for non eBPF install as the full packaged binary bundle (gz.run) installs fine. Was hoping this will work on Amazon Linux 2.

**Comment by thiagoftsm** [comment]
> cat /proc/version
> Linux version 4.14.154-128.181.amzn2.x86_64 (mockbuild@ip-10-0-1-129) (gcc version 7.3.1 20180712 (Red Hat 7.3.1-6) (GCC)) #1 SMP Sat Nov 16 21:49:00 UTC 2019

Thank you for the information.
@prologic we will need to bring another RHEL version for our Repository, right now we are not considering this RHEL flavor.

**Comment by prologic** [comment]
> Yea guess I will have to settle down for non eBPF install as the full packaged binary bundle (gz.run) installs fine. Was hoping this will work on Amazon Linux 2.

We're working on expanding support for `netdata-static` (kickstart-static6) + eBPF -- But its honestly actually very early stages and we haven't even merged the PR yet! We have to get this working on the systems we know about and fix bugs before we can even consider other more esoteric ones like AmazonLinux :)

**Comment by gamefundas** [comment]
Most companies that I know about who are AWS shoppers, are using Amazon Linux extensively mainly due to, good support, frequent security patches and healthy ecosystem that it offers. In my opinion this should be prioritized sooner.

**Comment by thiagoftsm** [comment]
Hi @gamefundas ,

We already had an internal talk about this situation and we will take a look on this ASAP.

Best regards!

**Comment by prologic** [comment]
@thiagoftsm Do you know what the status of this is? Can you help me understand where we're at?

**Comment by thiagoftsm** [comment]
Hi @prologic ,

I already could get with our team  the access key for we access the Amazon Linux, so we will have conditions to bring support for Amazon Linux soon.

**Comment by prologic** [comment]
> Hi @prologic ,
> 
> I already could get with our team the access key for we access the Amazon Linux, so we will have conditions to bring support for Amazon Linux soon.

Ahh brilliant. Is this really something you want to be doing? The Agent SRE team can probably take care of ensuring we have first class support for Amazon Linux by spinning up an EC2. Is this what you were planning on doing? We could also go an extra step and setup some CI with a self-hosted Github Actions runner on an Amazon Linux EC2.

**Comment by thiagoftsm** [comment]
Yes @prologic , I intend to bring this together the static compilation.

**Comment by thiagoftsm** [comment]
Hello @gamefundas ,

Today I followed the instructions present in this [link](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/amazon-linux-2-virtual-machine.html) that Amazon give us to run the AWS Linux 2. 

After to set up the VM, I installed Netdata, to do this firstly I install the following packages:

```
# yum install autoconf automake curl gcc git libmnl-devel libuuid-devel openssl-devel libuv-devel lz4-devel Judy-devel make nc pkgconfig python zlib-devel cmake
```

After this I got the Netdata branch from github and I compiled it with the following command:

```
./netdata-installer.sh --dont-wait --dont-start-it
```

The lastest argument `--dont-start-it` was given, because I was hoping that the `ebpf.plugin` would fail to start and I wanted to do some tests with it, but on the first run I got everything running:

```
netdata   2384  0.2  4.9 250648 37168 ?        Ssl  22:27   0:02 /usr/sbin/netdata -P /var/run/netdata/netdata.pid -D
netdata   2413  0.0  0.4  59256  3052 ?        Sl   22:27   0:00 /usr/sbin/netdata --special-spawn-server
netdata   2534  0.1  3.6 175708 27456 ?        Sl   22:27   0:01 /usr/bin/python /usr/libexec/netdata/plugins.d/python.d.plugin 1
netdata   2535  0.3  0.7  55816  5400 ?        S    22:27   0:03 /usr/libexec/netdata/plugins.d/apps.plugin 1
root      2536  0.1  0.5 215920  4416 ?        Sl   22:27   0:01 /usr/libexec/netdata/plugins.d/ebpf.plugin 1
netdata   2538  0.0  2.4 725128 18640 ?        Sl   22:27   0:00 /usr/libexec/netdata/plugins.d/go.d.plugin 1
root      3158  0.0  0.1 119420   948 tty2     S+   22:43   0:00 grep --color=auto netdata
```

And when I ran a `grep ebpf /var/log/netdata/error.log` I got the following expected output:

```
2020-06-28 22:27:29: netdata INFO  : PLUGINSD[ebpf] : thread created with task id 2525
2020-06-28 22:27:29: netdata INFO  : PLUGINSD[ebpf] : set name of thread 2525 to PLUGINSD[ebpf]
2020-06-28 22:27:29: netdata INFO  : PLUGINSD[ebpf] : connected to '/usr/libexec/netdata/plugins.d/ebpf.plugin' running on pid 2536
2020-06-28 22:27:29:  ERROR : MAIN : Does not have a configuration file inside `/etc/netdata/ebpf.conf. It will try to load stock file. (errno 2, No such file or directory)
2020-06-28 22:27:29: ebpf.plugin INFO  : EBPF SOCKET : thread created with task id 2559
2020-06-28 22:27:29: ebpf.plugin INFO  : EBPF SOCKET : set name of thread 2559 to EBPF SOCKET
2020-06-28 22:27:29: ebpf.plugin INFO  : EBPF SOCKET : Cannot load library /usr/libexec/netdata/plugins.d/libnetdata_ebpf.so.4.14.181-140.257.amzn2.x86_64 for the current kernel.
2020-06-28 22:27:29: ebpf.plugin INFO  : EBPF PROCESS : thread created with task id 2558
2020-06-28 22:27:29: ebpf.plugin INFO  : EBPF PROCESS : set name of thread 2558 to EBPF PROCESS
2020-06-28 22:27:29: ebpf.plugin INFO  : EBPF SOCKET : Default shared library /usr/libexec/netdata/plugins.d/libnetdata_ebpf.so.4.14.171 loaded with success.
2020-06-28 22:27:29: ebpf.plugin INFO  : EBPF SOCKET : The eBPF program socket was loaded with success.
2020-06-28 22:27:29: ebpf.plugin INFO  : EBPF PROCESS : Current shared library /usr/libexec/netdata/plugins.d/libnetdata_ebpf.so.4.14.171 loaded with success.
2020-06-28 22:27:29: ebpf.plugin INFO  : EBPF PROCESS : The eBPF program process was loaded with success.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.tcp_functions/main.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.tcp_functions/tcp_sendmsg.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.tcp_functions/tcp_cleanup_rbuf.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.tcp_functions/tcp_close.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.tcp_bandwidth/main.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.tcp_bandwidth/tcp_sendmsg.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.tcp_bandwidth/tcp_cleanup_rbuf.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.tcp_bandwidth/tcp_close.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.udp_functions/main.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.udp_functions/udp_sendmsg.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.udp_functions/udp_recvmsg.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.udp_bandwidth/main.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.udp_bandwidth/udp_sendmsg.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.udp_bandwidth/udp_recvmsg.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.file_descriptor/main.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.file_descriptor/do_sys_open.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.file_descriptor/__close_fd.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.deleted_objects/main.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.deleted_objects/vfs_unlink.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.io/main.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.io/vfs_read.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.io/vfs_write.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.io_bytes/main.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.io_bytes/vfs_read.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.io_bytes/vfs_write.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.process_thread/main.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.process_thread/_do_fork.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.process_thread/sys_clone.db.
2020-06-28 22:27:30: netdata INFO  : PLUGINSD[ebpf] : Initializing file /var/cache/netdata/ebpf.exit/main.db.
```

I have few questions for you, considering that I could not create the problem you are describing:

1 - Did you try to use `ebpf.plugin` with the latest Netdata  version 1.23.0?
2 - What is the output for the command `uname -a`? I got the following
```
Linux aws_test 4.14.181-140.257.amzn2.x86_64 #1 SMP Wed May 27 02:17:36 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```  

Best regards!

**Comment by cakrit** [comment]
No response since August, closing. If there's additional info, please comment and we can reopen it.
